src/applications/repository/engine/PhabricatorRepositoryPullEngine.php

```diff
@@ -266,6 +266,15 @@ private function executeGitCreate() {
     $repository->execxRemoteCommand(
       'init --bare -- %s',
       $path);
+
+    // Check if repository fetching should use --shallow-exclude.
+    $config = PhabricatorEnv::getEnvConfig('diffusion.shallow-repos');
+    $callsign = $repository->getCallsign();
+    if (array_key_exists($callsign, $config)) {
+      $cutoff = $config[$callsign];
+      $repository->execLocalCommand(
+        'config --local --add "alias.ft" "fetch --shallow-since=%s"', $cutoff);
+    }
```

Comment:
This is here so that we can check the config only once (upon creation) rather than every time we are fetching.